skills/csharp/tests/musicskill.Tests/PlayMusicDialogTests.cs

```diff
@@ -0,0 +1,108 @@
+                .Send(PlayMusicDialogUtterances.PlayMusic)
+                .AssertReply((activity) =>
+                {
+                    var a = (Activity)activity;
+                    Assert.AreEqual(ActivityTypes.Event, a.Type);
+                    Assert.AreEqual(typeof(OpenDefaultApp), a.Value.GetType());
+                    Assert.AreEqual((a.Value as OpenDefaultApp).MusicUri, "spotify:playlist:37i9dQZF1DXcCnTAt8Cf");
+                })
+                .AssertReplyOneOf(ParseReplies(MainResponses.CompletedMessage))
+                .StartTestAsync();
+        }
+
+        [TestMethod]
+        public async Task Test_Sample_Action()
+        {
+            await GetSkillTestFlow()
+               .Send(new Activity(type: ActivityTypes.Event, name: Events.PlayMusic))
+               .AssertReplyOneOf(ParseReplies(MainResponses.NoResultstMessage))
+               .AssertReply((activity) =>
+               {
+                   var a = (Activity)activity;
+                   Assert.AreEqual(ActivityTypes.EndOfConversation, a.Type);
+                   Assert.AreEqual(typeof(ActionResult), a.Value.GetType());
+                   Assert.AreEqual((a.Value as ActionResult).ActionSuccess, false);
+               })
+               .StartTestAsync();
+        }
+
+        [TestMethod]
+        public async Task Test_Sample_Action_w_Input()
+        {
+            var actionInput = new SearchInfo() { MusicInfo = "music" };
+
+            await GetSkillTestFlow()
+               .Send(new Activity(type: ActivityTypes.Event, name: Events.PlayMusic, value: JObject.FromObject(actionInput)))
+                .AssertReply((activity) =>
+                {
+                    var a = (Activity)activity;
+                    Assert.AreEqual(ActivityTypes.Event, a.Type);
+                    Assert.AreEqual(typeof(OpenDefaultApp), a.Value.GetType());
+                    Assert.AreEqual((a.Value as OpenDefaultApp).MusicUri, "spotify:playlist:37i9dQZF1DXcCnTAt8Cf");
+                })
+               .AssertReply((activity) =>
+               {
+                   var a = (Activity)activity;
+                   Assert.AreEqual(ActivityTypes.EndOfConversation, a.Type);
+                   Assert.AreEqual(typeof(ActionResult), a.Value.GetType());
+                   Assert.AreEqual((a.Value as ActionResult).ActionSuccess, true);
+               })
+               .StartTestAsync();
+        }
+
+        [TestMethod]
+        public async Task Test_Sample_Dialog_SkillMode()
+        {
+            await GetSkillTestFlow()
+               .Send(PlayMusicDialogUtterances.PlayMusic)
+                .AssertReply((activity) =>
+                {
+                    var a = (Activity)activity;
+                    Assert.AreEqual(ActivityTypes.Event, a.Type);
+                    Assert.AreEqual(typeof(OpenDefaultApp), a.Value.GetType());
+                    Assert.AreEqual((a.Value as OpenDefaultApp).MusicUri, "spotify:playlist:37i9dQZF1DXcCnTAt8Cf");
```

Comment:
Put this "spotify:playlist:37i9dQZF1DXcCnTAt8Cf" to a field? Bacause it's used many times and is a little ugly hardcode here.

Reply:
sure, moved